apps/mux/src/index.tsx

```diff
@@ -0,0 +1,404 @@
+  };
+
+  pollForAssetDetails = async () => {
+    if (!this.state.value || !this.state.value.assetId) {
+      throw Error(
+        'Something went wrong, because by this point we require an assetId.'
+      );
+    }
+
+    const asset = await this.getAsset();
+
+    if (!asset) {
+      throw Error('Something went wrong, we were not able to get the asset.');
+    }
+
+    await this.props.sdk.field.setValue({
+      uploadId: this.state.value.uploadId,
+      assetId: this.state.value.assetId,
+      playbackId: asset['playback_ids'][0].id,
+      ready: asset.status === 'ready',
+      ratio: asset.ratio,
+    });
+
+    if (asset.status !== 'ready') {
+      await delay(500);
+      await this.pollForAssetDetails();
+    }
+  };
+
+  onPlayerReady = () => this.props.sdk.window.updateHeight();
+
+  render = () => {
+    if (this.state.error) {
+      return (
+        <Note noteType="negative">
+          <span>{this.state.error}</span>
+          <span>
+            {this.state.errorShowResetAction ? (
+              <Button
+                buttonType="negative"
+                size="small"
+                onClick={this.resetField}
+                className="reset-field-button"
+              >
+                Reset this field
+              </Button>
+            ) : null}
+          </span>
+        </Note>
+      );
+    }
+
+    if (this.state.isDeleting) {
+      return (
+        <Paragraph>
+          <Spinner size="small" /> Deleting this asset.
+        </Paragraph>
+      );
+    }
+
+    if (this.state.value) {
+      if (this.state.value.assetId && !this.state.value.ready) {
+        return (
+          <Paragraph>
+            <Spinner size="small" /> Waiting for asset to be playable!
+          </Paragraph>
+        );
+      }
+
+      if (this.state.value.ready) {
+        return (
+          <div>
+            <Player
+              playbackId={this.state.value.playbackId}
+              ratio={this.state.value.ratio}
+              onReady={this.onPlayerReady}
+            />
+            {this.state.value.assetId ? (
+              <DeleteButton requestDeleteAsset={this.requestDeleteAsset} />
+            ) : null}
+          </div>
+        );
+      }
+    }
+
+    if (this.state.uploadProgress) {
+      return (
+        <div>
+          <Paragraph>
+            <Spinner size="small" />{' '}
+            {this.state.uploadProgress < 100
+              ? 'Uploading file to Mux...'
+              : 'Upload complete! Waiting for created asset details...'}
+          </Paragraph>
+          <div
+            className="progress"
+            style={{ width: `${this.state.uploadProgress}%` }}
+          />
+        </div>
+      );
+    }
+
+    return (
+      <input type="file" className="cf-file-input" onChange={this.onChange} />
+    );
+  };
+}
+
+init(sdk => {
+  const Component = sdk.location.is(locations.LOCATION_APP_CONFIG) ? Config : App;
+  render(
+    <Component sdk={sdk as FieldExtensionSDK} />,
```

Comment:
Since you have ternary logic above, the type of the SDK won't be `FieldExtensionSDK` on the CONFIG location.

Reply:
I used `BaseExtensionSDK` as the type that gets passed to `Config`  - seemed like the most appropriate one to me, let me know if there's a better option please!